src/atoms/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,73 @@
+import { padding } from 'polished';
+import React, { Component, createRef, ReactNode } from 'react';
+
+import styled from '_styled-components';
+import { borderRadius, scale } from 'Theme';
+import triangle from './tooltip-triangle.svg';
+
+const Relative = styled.div`
+  position: relative;
+`;
+
+const Absolute = styled.div<{ height?: number }>`
+  position: absolute;
+  bottom: ${props => props.height}px;
+`;
+
+const Box = styled.div`
+  border: 0.0625em solid #bac9d4;
+  background: ${props => props.theme.controlBackground};
+  border-radius: ${borderRadius};
+  box-shadow: 0 0.1875em 0.375em 0 #00000012,
+    0 0.4375em 0.625em 0.4375em #32325d1a;
+  ${padding(scale(-1))};
+`;
+
+const Triangle = styled.img`
+  position: relative;
+  top: -0.25em;
+  left: 50%;
+  transform: translateX(-50%);
+`;
+
+Triangle.defaultProps = { src: triangle };
+
+export class Tooltip extends Component<{
+  tooltip: ReactNode;
+  children(props: Record<'onMouseOver' | 'onMouseOut', () => void>): ReactNode;
```

Comment:
Everything looks solid! I'd appreciate an explanation on what's going on here specifically so I can learn.

Reply:
This is a function as child. The children prop is a function that gets handlers passed to it so you can spread them onto any element to make it hoverable. Maybe we can use a simmpler API, but I thought this would be the most flexible so we aren't forcing users to render specific tags.

This line is shorthand for this type:
```ts
children: (props: { onMouseOver: () => void; onMouseOut: () => void }) => ReactNode;
```

Reply:
Thanks for the explanation. This makes sense and seems great for flexibility.